caas/kubernetes/provider/bootstrap.go

```diff
@@ -40,6 +42,39 @@ var (
 	TemplateFileNameAgentConf = "template-" + agent.AgentConfigFilename
 )
 
+type controllerServiceSpec struct {
+	// serviceType is required.
+	serviceType core.ServiceType
+	// annotations is optional.
+	annotations k8sannotations.Annotation
+}
+
+var controllerServiceSpecs = map[string]*controllerServiceSpec{
```

Comment:
There should be a default as well for clouds we don't specifically enumerate below